misc/vtk/src/processors/vtktovolume.cpp

```diff
@@ -0,0 +1,422 @@
+        for (auto *prop : dataArrays_.getProperties()) {
+            BoolProperty *boolProp = dynamic_cast<BoolProperty *>(prop);
+            if (boolProp && boolProp->get()) allSelected << boolProp->getDisplayName() << ';';
+        }
+    }
+    s.serialize("selectedArrays", allSelected.str(), SerializationTarget::Attribute);
+}
+
+void VTKtoVolume::VTKArrayList::addArray(const std::string &name, int numChannels, void *dataPtr) {
+    auto boolProp = new BoolProperty(name, name + " (" + std::to_string(numChannels) + ')', false);
+    addProperty(boolProp);
+    numArrayComponents.push_back(numChannels);
+    ptrArrayData.push_back(reinterpret_cast<unsigned char *>(dataPtr));
+}
+
+void VTKtoVolume::VTKArrayList::clear() {
+    std::vector<std::string> identifiers{};
+    for (const auto &property : getProperties()) {
+        identifiers.emplace_back(property->getIdentifier());
+    }
+    for (const auto &identifier : identifiers) {
+        Property *prop = removeProperty(identifier);
+        delete prop;
+    }
+    numArrayComponents.clear();
+    ptrArrayData.clear();
+}
+
+void VTKtoVolume::updateFormats() {
+    NetworkLock lock;
+    reactToChange_ = false;
+
+    DataFormatId formerFormat = DataFormatId::NotSpecialized;
+    if (dataFormats_.size()) formerFormat = dataFormats_.getSelectedValue();
+
+    dataFormats_.clearOptions();
+    if (!inport_.hasData()) {
+        dataArrays_.clear();
+        return;
+    }
+
+    // Get dimensions. If none are given, it's not a rectilinear grid.
+    const VTKDataSet &vtkData = *inport_.getData();
+    std::optional<size3_t> dimensions = inport_.getData()->getDimensions();
+    if (!dimensions) {
+        LogWarn("Input vtk data set is not rectilinear.");
+        return;
+    }
+
+    // Get data attributes.
+    auto cellData = vtkData->GetCellData();
+    auto pointData = vtkData->GetPointData();
+    if ((useCellData_.get() && !cellData) || (!useCellData_.get() && !pointData)) return;
+    int numArrays =
+        useCellData_.get() ? cellData->GetNumberOfArrays() : pointData->GetNumberOfArrays();
+
+    // Find first non-internal data array and take that value type.
+    std::vector<int> insertedValues;
+    for (int i = 0; i < numArrays; ++i) {
+        auto array = useCellData_.get() ? cellData->GetArray(i) : pointData->GetArray(i);
+        int vtkTypeId = array->GetDataType();
+
+        if (std::find(insertedValues.begin(), insertedValues.end(), vtkTypeId) !=
+            insertedValues.end())
+            continue;
+
+        const DataFormatBase *format =
+            vtkutil::getFormatFromVtkId(vtkTypeId, array->GetElementComponentSize() * 8);
+        if (!format) continue;
+
+        insertedValues.push_back(vtkTypeId);
+        dataFormats_.addOption(format->getString(), format->getString(), format->getId());
+    }
+
+    // Try to keep arrays if the new input has arrays of the same type.
+    bool keep = dataFormats_.setSelectedValue(formerFormat);
```

Comment:
Why not always try to keep the settings?

Reply:
The idea is to keep the settings when we get similar data (i.e., after deserialization or when getting another time-step/member) with the same data array setup.
So when `float` was selected as the format to use before, I would not expect to find the same array names in a data set that does not contain any float data.
But you are right, we could just try to find arrays with the same names anyway.

Reply:
Now the processor always tries to keep the settings, and will trigger a conversion automatically iff all arrays could be carried over.